starport/pkg/truncatedbuffer/truncatedbuffer_test.go

```diff
@@ -0,0 +1,37 @@
+package truncatedbuffer
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriter(t *testing.T) {
+	ranBytes10 := make([]byte, 10)
+	rand.Read(ranBytes10)
+	ranBytes1000 := make([]byte, 1000)
+	rand.Read(ranBytes1000)
+
+	// TruncatedBuffer has a max capacity
+	b := NewTruncatedBuffer(100)
+
+	require.Equal(t, 100, b.GetCap())
+
+	n, err := b.Write(ranBytes10)
+	require.NoError(t, err)
+	require.Equal(t, 10, n)
+	require.Equal(t, ranBytes10, b.GetBuffer().Bytes())
+
+	n, err = b.Write(ranBytes1000)
```

Comment:
The behavior of `TruncatedBuffer` seems like unexpected to me. The bytes written into the buffer never gets overwritten once buffer is full.

Instead, previously written bytes should be discarded in FIFO order as buffer starts overflowing and for this line, last `100` bytes of `1000` should be residing in the buffer. I think the expected behavior would be like this:

1- Let's say that we have initialized `TruncatedBuffer` with cap `5`. In this case buffer should look like below:
```
[0,0,0,0,0]
// ~ 5 seats left.
```

1- After writing `1,2,3`:
```
[1,2,3,0,0]
// 2 seats left.
```

2- After writing `4,5`:
```
[1,2,3,4,5]
// no seats left.
```

3- After writing `6,7`:
```
[3,4,5,6,7]
// the algorithm is basically to shift one byte from the buffer to remove the first byte and 
// have an empty seat in the end. then write the next given byte which is `6`, 
// and repeat shifting and writing X times untill X=0.
// (initally X=len([6,7]).
```

4- After writing `8,9,10,11,12,13`:
```
[9,10,11,12,13]
```

Reply:
This is on purpose. This is because for a start error we are only interested in the first bytes.

When creating a package for this it should become more universal and not only related to this specific use case but for me, it still makes sense to have this behavior because the name "truncated" means that the end is removed. With the FIFO behavior, I would rather name it something like `FixSizedBuffer`

One problem is I wanted to return n=0 when no bytes are written because max size is reached which would make more sense but when doing this the program was stopping once the cap was reached

Reply:
Alright, this is good for me.